### PR TITLE
[Merged by Bors] - fix(Tactic/Linter): `upstreamableDecl` should treat `structure`s like `def`s

### PR DESCRIPTION
**Pull request by Vierkantor**
Reported on Zulip: https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/topic/upstreamableDecl.20linter.20bug/near/483790949

We want to treat a `structure`/`inductive` declaration in the current file just like a `def` for the `upstreamableDecl` linter: potentially place a warning on the `structure` itself, but otherwise don't place warnings on downstream uses.

There's a small complication where `inductive` declarations are reported to depend on their own constructors. I couldn't find a good way to determine whether any given constant is indeed a constructor declared in some given piece of syntax, so instead we allow depending on constructors just like we allow depending on theorems.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Tactic/Linter/UpstreamableDecl.lean [hunk 1]
```diff
@@ -21,8 +21,12 @@ def Lean.Name.isLocal (env : Environment) (decl : Name) : Bool :=
 
 open Mathlib.Command.MinImports
 
-/-- Does the declaration with this name depend on `def`s in the current file? -/
-def Lean.Environment.localDefDependencies (env : Environment) (stx id : Syntax) :
+/-- Does the declaration with this name depend on definitions in the current file?
+
+Here, "definition" means everything that is not a theorem, and so includes `theorem`,
```
Comment (adomani) [inline Mathlib/Tactic/Linter/UpstreamableDecl.lean:26]:
```suggestion
Here, "definition" means everything that is not a theorem, and so includes `def`,
```
I guess!  :smile:

**Review by adomani — APPROVED** [review]
Looks good to me!

maintainer delegate?

@kim-em in case you want to take a look!

Oops, I did not realize that there was some shaking to be done!

File: Mathlib/Tactic/Linter/UpstreamableDecl.lean (hunk 1 above)
Comment (Vierkantor) [reply Mathlib/Tactic/Linter/UpstreamableDecl.lean]: You are absolutely right of course, well spotted :D

**Comment by kim-em** [comment]
bors merge

**Pull request closed without merging**